### ISSUE TITLE
bug: redundant instructions in KubeStellar Documentation

### ISSUE THREAD

**Issue by rchen-devv** [kubestellar/kubestellar#2306]
### Describe the bug

When looking for resources to how to use binding policy on a cluster, there are multiple pages that accomplish the same task.  For example, the new release 0.23.1 has instructions that accomplish the same thing as release 0.23.0. Another example of the redundant instruction is that the site management is more disorganized in release-0.23.1. The "quick start" and "user guide" tab on the top brings you to the "user guide". 

Example 1: 
![Screenshot 2024-07-10 at 12 51 24 AM](https://github.com/kubestellar/kubestellar/assets/169481903/eef4de30-da87-4a05-ab2e-4ca7a18edbb0)
Is the same as 
![Screenshot 2024-07-10 at 12 52 37 AM](https://github.com/kubestellar/kubestellar/assets/169481903/50f04130-ae3b-40f6-8d84-7ea150308cb5)
just with some additional information but idk why the page get a update instead of constantly making new releases for small changes. 

Example 2: 
![Screenshot 2024-07-10 at 12 53 46 AM](https://github.com/kubestellar/kubestellar/assets/169481903/77e2356a-eb84-443e-8e6d-b93f74d4653f)
brings me to the same domain as "user guide"
![Screenshot 2024-07-10 at 12 54 42 AM](https://github.com/kubestellar/kubestellar/assets/169481903/139e7dfd-959c-4fc7-9b77-faae97ba656e)

I feel like these small instances in these documentations brings a bunch of confusion to a new user. 

### Steps To Reproduce

Example 1: 
Go to his site right here: 
[https://docs.kubestellar.io/release-0.23.0/direct/examples/](https://docs.kubestellar.io/release-0.23.0/direct/examples/)
and here is the second site where it has the same information but accomplish the same thing 
[https://docs.kubestellar.io/release-0.23.1/direct/user-guide-intro/](https://docs.kubestellar.io/release-0.23.1/direct/user-guide-intro/)
Example 2:
go to this site right here 
[https://docs.kubestellar.io/release-0.23.1/direct/user-guide-intro/](https://docs.kubestellar.io/release-0.23.1/direct/user-guide-intro/)
click on "user guide" and "quick start" you can see that the information for quick start is in the domain of user guide. 

### Expected Behavior

Example 1: Just a suggestion to clean up redundant documentation so users are not confused about what each release is accomplishing
Example 2: Get rid of quick start as a tab or move quick start to another domain that is not tied to "user guide"

### Want to contribute?

- [ ] I would like to work on this issue.

**Comment by clubanderson** [comment]
@MikeSpreitzer can we remove old versions of 0.23 and keep only the latest of that major?

**Comment by MikeSpreitzer** [comment]
@rchen-devv: we are doing incremental development, so each release has a lot in common with the previous release and the next release. Is this surprising or confusing?

**Comment by MikeSpreitzer** [comment]
Perhaps your first point is that the change from 0.23.0 to 0.23.1 was not worth making a release. We chose to do that because there were significant changes to the organization of the documentation, and wanted to publish the improved documentation. We were getting complaints about the website content for 0.23.0.

We have continued to work on the web site. The latest in `main` can be seen at https://docs.kubestellar.io/unreleased-development/ . Comments are encouraged.

**Comment by rchen-devv** [comment]
@MikeSpreitzer I understand that the new release of 0.23.1 has the most up to date instruction but what I don't get is why is the old instructions still present? If the new instructions of release-0.23.1 is the most up to date don't we want to at least archive the older version to prevent confusion to the user?

**Comment by rchen-devv** [comment]
@MikeSpreitzer for the "Quick Start" and "User Guide" it is indeed two different sites but why is there two links to the same domain? 
<img width="291" alt="Screenshot 2024-07-10 at 2 15 13 PM" src="https://github.com/kubestellar/kubestellar/assets/169481903/8dbd6c60-d121-4638-9992-c91bbe7868ca">
for example in this screenshot you can click on either  "Quick Start" and "User Guide" and it brings you to the same domain instead of separate. My suggestion is to either remove quick start from the tab or move the quick start content to another domain that is not user guide to prevent redundant links

**Comment by MikeSpreitzer** [comment]
A release is a self-contained self-consistent not-necessarily-consistent-with-any-other thing. The website is _versioned_ . Each version of the website is for a particular release, except the `unreleased-development` version, which is for the `main` branch. The `release-0.23.1` version of the website is for release 0.23.1; the `release-0.23.0` version of the website is for release 0.23.0; the `release-0.22.0` version of the website is for release 0.22.0; and so on.

We _could_ delete the `release-0.23.0` version of the website, but that would be part of dropping support for release 0.23.0. I would be OK with that, but it should be understood and agreed that this is what is being done. And it would not remove any other version of the website.

**Comment by MikeSpreitzer** [comment]
We do not have a concept of "domains" within the website. The highest division within the website is between _versions_. Within a version, we have navigation among pages, via links across the top (if the browswer is wide enough for those to appear there) and links listed in a panel on the left (if the browswer is wide enough for it to show up there).

**Comment by MikeSpreitzer** [comment]
@clubanderson: I am getting the idea that the version and navigation structure of MkDocs and `mike` are what the OP did not get.

**Comment by clubanderson** [comment]
@MikeSpreitzer well, I can see @rchen-devv 's point. It seems a bit unnatural to have 'user guide' AND 'quick start' as tabs that point at the same menu. Maybe consolidation of that is useful, or another tree altogether for 'user guide' outside of the quickstart bits.  WDYT

**Comment by MikeSpreitzer** [comment]
Yes, I can imagine a reader drawing the implication that the things listed across the top are disjoint. @KPRoche , what do you think?

**Comment by KPRoche** [comment]
I made the choice to add the redundant links in the masthead during the redesign.

Calling out the Quick Start as a separate top level link that jumps to the Quick Start subsection of the User Guide was a *deliberate* redundancy I added, based on years of observing web user behavior.

Redundant links to the same page recognize that different users look for info in different ways. An example I have run into many times when building conference websites, for instance, is one subset of users looks for "registration" and another subset looks for "membership" to sign up. Neither are wrong, but neither think of looking for the other term and so without the redundancy, half the visitors are frustrated.

Many open source programs, including some of KubeStellar's "competitors" feature a Quick Start or Demo link for visitors immediately interested trying the program out. That is the reason for the Quick Start link in the masthead.

The *information* on that page is an abbreviated setup process, which _absolutely_ falls under the purview of what should be in a user guide.

In a printed user manual, there is often a "Getting Started" chapter with a quick demo the reader can try, followed by a "Operation" or "General Use" chapter which expands on the capabilities and options for the tool in question. This is the same paradigm.
The Quick Start link is just a extra navigation shortcut to satisfy the people who will be looking for one; the information it takes you to is a logical part of our larger User Guide.

**Comment by clubanderson** [comment]
@KPRoche why not make a separate tree for user guide?

**Comment by KPRoche** [comment]
> @KPRoche why not make a separate tree for user guide?

The way mkdocs works weird things happen in the displayed navigation if you use the same file in different trees, or even multiply in one tree. I've observed that when you follow a link to such a file, the displayed left menu will always be for the first time the file is processed during the generation of the site, even if you used a link in a different part of the menus to get there.

To avoid that and build an independent tree, we'd need multiple files. 
We certainly don't want to maintain multiple copies of the same information in different files. Letting the navigation help people find it is the point.
If this were html rather than mkdocs, I would encode hover text for the links, so hovering over "Quick Start" would display "Takes you directly to the Quick Start section of the User Guide"

**Comment by KPRoche** [comment]
And more fundamentally, _how is the quick start/getting started/demo instructions NOT appropriate to be a chapter of the User Guide?_

**Comment by MikeSpreitzer** [comment]
We _have_ removed version `release-0.23.0` from the website --- despite the fact that @cwiklik is still using it. We have removed many versions. But still the website remains something that has multiple versions. I was expecting that readers can get the idea that the website has multiple versions corresponding to multiple releases. This is a structure that is common to many projects.

**Comment by clubanderson** [comment]
Closing as stale. This bug about redundant instructions from July 2024 has not seen activity. The documentation has been significantly restructured since then. If there are current redundancy issues, please open a new issue with specifics.
